packages/synapse-react-client/src/lib/containers/markdown/MarkdownEditor.tsx

```diff
@@ -67,8 +67,9 @@ export const MarkdownEditor: React.FunctionComponent<MarkdownEditorProps> = ({
     setTagModalWithKeyboard(false)
   }
 
-  const handleTagModal = (e: React.KeyboardEvent) => {
-    if (e.key == '@') {
+  const handleTagModal = (text: string) => {
+    const start = textAreaRef.current && textAreaRef.current.selectionStart
+    if (start && start > 0 && text.charAt(start - 1) === '@') {
```

Comment:
Handle case where `@` is pressed in the middle of text -- 

https://user-images.githubusercontent.com/26949006/236282442-43739f66-6658-4dde-aa5d-922cabb3d1f6.mov